### PR TITLE
ac:u IsConnected implemented

### PR DESCRIPTION
**Pull request by mailwl**
code from: Heroes of the ruin - Hangs at title screen #1157
also fix: Project X Zone 2 - Unimplemented StoreDataCache #1428

### CODE REVIEW

File: src/core/hle/service/ac_u.cpp [hunk 1]
```diff
@@ -28,6 +28,21 @@ static void GetWifiStatus(Service::Interface* self) {
     LOG_WARNING(Service_AC, "(STUBBED) called");
 }
 
+/**
+ * AC_U::IsConnected service function
```
Comment (bunnei) [inline src/core/hle/service/ac_u.cpp]: What is this actually used for?

Comment (mailwl) [inline src/core/hle/service/ac_u.cpp]: i think, if we don't set cmd_buff[2] to 0, then it contains garbage (not zero), and a game think, the service is connected, and wait something from it forever.
What exactly AC service is - i have no idea

Comment (yuriks) [inline src/core/hle/service/ac_u.cpp]: Pretty obviously Wifi access point stuff: https://www.3dbrew.org/wiki/AC_Services

File: src/core/hle/service/ac_u.cpp [hunk 2]
```diff
@@ -28,6 +28,21 @@ static void GetWifiStatus(Service::Interface* self) {
...
+ *  Outputs:
+ *      1 : Result of function, 0 on success, otherwise error code
+ *      2 : bool, is connected
+ */
+static void IsConnected(Service::Interface* self) {
+    u32* cmd_buff = Kernel::GetCommandBuffer();
+
+    cmd_buff[1] = 0; // No error
```
Comment (yuriks) [inline src/core/hle/service/ac_u.cpp]: use RESULT_SUCCESS.raw instead of 0.

**Comment by yuriks** [comment]
Change commit message to "ac:u: Stub IsConnected"

LGTM apart from the two nits, though I haven't verified that the actual buffer format is correct.

**Comment by mailwl** [comment]
about buffer format. How it call in Heroes of Ruins: parameters zero and pointer to local var in stack:
```
.text:002BEAA8                 MOV             R0, #0
.text:002BEAAC                 MOV             R1, SP
.text:002BEAB0                 STRB            R0, [SP,#0x10+var_10]
.text:002BEAB4                 BL              IsConnected
.text:002BEAB8                 LDRSB           R0, [SP,#0x10+var_10
```
the function itself:
```c++
int __fastcall IsConnected(int a1, _BYTE *a2)
{
  char *v2; // r4@1
  _DWORD *cmd_buff; // r4@1
  int result; // r0@1

  v2 = (char *)__mrc(15, 0, 13, 0, 3);
  *((_DWORD *)v2 + 0x20) = 0x3E0042;
  cmd_buff = v2 + 0x80;
  cmd_buff[1] = a1;
  cmd_buff[2] = 0x20;
  result = dword_58484C;
  __asm { SVC             0x32 }
  if ( (dword_58484C & 0x80000000 & 0x80000000) == 0 )
  {
    *a2 = *((_BYTE *)cmd_buff + 8); // cmd_buff[2] low byte
    result = cmd_buff[1];
  }
  return result;
}
```

**Pull request merged**